test/system/src/test/java/io/pravega/test/system/AbstractFailoverTests.java

```diff
@@ -173,24 +185,36 @@ void performFailoverForTestsInvolvingTxns() {
                 testState.eventReadCount.get(),  testState.eventWriteCount.get());
 
         //Scale down segment store instances to 2
-        segmentStoreInstance.scaleService(2, true);
+        try {
+            Exceptions.handleInterrupted(() -> segmentStoreInstance.scaleService(2).get());
+        } catch (ExecutionException e) {
+            throw new TestFrameworkException(RequestFailed, "Scaling operation failed", e);
+        }
         //zookeeper will take about 30 seconds to detect that the node has gone down
         Exceptions.handleInterrupted(() -> Thread.sleep(WAIT_AFTER_FAILOVER_MILLIS));
         log.info("Scaling down Segment Store instances from 3 to 2");
         log.info("Read count: {}, write count: {} after Segment Store  failover after sleep",
                 testState.eventReadCount.get(),  testState.eventWriteCount.get());
 
         //Scale down controller instances to 2
-        controllerInstance.scaleService(2, true);
+        try {
+            Exceptions.handleInterrupted(() -> controllerInstance.scaleService(2).get());
+        } catch (ExecutionException e) {
```

Comment:
Here and elsewhere in this file FutureHelpers.getAndHandleExceptions() would do this more concisely.

Reply:
done.